src/meta/src/hummock/compaction_scheduler.rs

```diff
@@ -161,15 +164,21 @@ where
         // 2. Assign the compaction task to a compactor.
         'send_task: loop {
             // 2.1 Select a compactor.
-            let compactor = match self.compactor_manager.next_compactor() {
+            let compactor = match self
```

Comment:
Could we check idle compactor before creating a new compact task? or the compact task may hold the files (make it as pending state) for a long time....

Reply:
> check idle compactor before creating a new compact task

It does help in common sense. But here compaction scheduler is single-threaded, so I think it does no harm to hold files.

Reply:
No. I means that, if we do not hold these files, in next selection (maybe in a soon time), we can give a better compaction strategy for compact task.

Reply:
Performing potential time consuming compactor selection beforehand also has a drawback: it may delay trivial move task. Because

- trivial move task is only identifiable after get_compact_task.
- trivial move task should not be blocked by compactor selection.